src/main/java/com/iota/iri/network/NeighborRouter.java

```diff
@@ -0,0 +1,840 @@
+
+        URI neighborURI;
+        try {
+            neighborURI = new URI(uri);
+        } catch (URISyntaxException e) {
+            log.error("URI {} raised URI Syntax Exception", uri);
+            return Optional.empty();
+        }
+        if (!isURIValid(neighborURI)) {
+            return Optional.empty();
+        }
+        return Optional.of(neighborURI);
+    }
+
+    /**
+     * Checks whether the given URI is valid. The URI is valid if it is not null and it uses TCP as the protocol.
+     * 
+     * @param uri The URI to check
+     * @return true if the URI is valid, false if not
+     */
+    public static boolean isURIValid(final URI uri) {
+        if (uri == null) {
+            log.error("Cannot read URI schema, please check neighbor config!");
+            return false;
+        }
+
+        if (!uri.getScheme().equals("tcp")) {
+            log.error("'{}' is not a valid URI schema", uri);
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Returns the {@link TransactionProcessingPipeline}.
+     * 
+     * @return the {@link TransactionProcessingPipeline} used by the {@link NeighborRouter}
+     */
+    public TransactionProcessingPipeline getTransactionProcessingPipeline() {
+        return txPipeline;
+    }
+
+    /**
+     * Gets all neighbors the {@link NeighborRouter} currently sees as either connected or attempts to build connections
+     * to.
+     * 
+     * @return The neighbors
+     */
+    public List<Neighbor> getNeighbors() {
+        List<Neighbor> neighbors = new ArrayList<>(connectedNeighbors.values());
+        reconnectPool.forEach(uri -> {
+            neighbors.add(new NeighborImpl<>(null, null, uri.getHost(), uri.getPort(), null));
+        });
+        return neighbors;
+    }
+
+    /**
+     * Gets the currently connected neighbors.
+     * 
+     * @return The connected neighbors.
+     */
+    public Map<String, Neighbor> getConnectedNeighbors() {
+        return connectedNeighbors;
+    }
+
+    /**
+     * Gossips the given transaction to the given neighbor.
+     *
+     * @param neighbor The {@link Neighbor} to gossip the transaction to
+     * @param tvm      The transaction to gossip
+     * @throws Exception thrown when loading a hash of transaction to request fails
+     */
+    public void gossipTransactionTo(Neighbor neighbor, TransactionViewModel tvm) throws Exception {
+        gossipTransactionTo(neighbor, tvm, false);
+    }
+
+    /**
+     * Gossips the given transaction to the given neighbor.
+     *
+     * @param neighbor     The {@link Neighbor} to gossip the transaction to
+     * @param tvm          The transaction to gossip
+     * @param useHashOfTVM Whether to use the hash of the given transaction as the requested transaction hash or not
+     * @throws Exception thrown when loading a hash of transaction to request fails
+     */
+    public void gossipTransactionTo(Neighbor neighbor, TransactionViewModel tvm, boolean useHashOfTVM)
```

Comment:
I think this should be a public method of `neighbor`.
I don't think the current classes that invoke this method should have access to the `NeighborRouter`

Reply:
I am reluctant to moving this method into `Neighbor` as it would unnecessarily add an additional dependency (TransactionRequester) to it. I think this is fine for now.

Reply:
I didn't notice `TransactionRequester`
I don't think the method belongs here but you are correct it shouldn't be in `Neighbor`. It is a model and shouldn't have dependencies carrying logic.
I guess it can stay here for now